packages/typeorm/src/TypeORMTransactionManager.ts

```diff
@@ -1,61 +1,109 @@
-import { Connection, EntityManager, getConnection, QueryFailedError } from 'typeorm';
-import { Operation, TransactionManager, TransactionOptions, TypeORMOptions } from '@leocode/nest-tx-core';
-import { isTypeORMTransaction, TypeORMTransaction } from './TypeORMTransaction';
-import { PostgresDriver } from 'typeorm/driver/postgres/PostgresDriver';
-import { NotATypeORMTransactionError } from './NotATypeORMTransactionError';
+import {
+  Connection,
+  EntityManager,
+  getConnection,
+  QueryFailedError,
+} from "typeorm";
+import {
+  Operation,
+  TransactionManager,
+  TransactionOptions,
+  TypeORMOptions,
+} from "@leocode/nest-tx-core";
+import { isTypeORMTransaction, TypeORMTransaction } from "./TypeORMTransaction";
+import { PostgresDriver } from "typeorm/driver/postgres/PostgresDriver";
+import { NotATypeORMTransactionError } from "./NotATypeORMTransactionError";
 
 const isRetriableError = (err: unknown, connection: Connection): boolean => {
   if (!(err instanceof QueryFailedError)) {
     return false;
   }
 
+  // NOTE: Seems that the typescript doesn't agree on changing the type of err to QueryFailedError,
+  // even if the logical operator for instanceof checkes clearly that if anything other than instanceof QueryFailedError,
+  // returns false.
+  const error = err as QueryFailedError;
+
   // POSTGRESQL SERIALIZATION FAILURE
-  if (connection.driver instanceof PostgresDriver && err.driverError.code === '40001') {
+  if (
+    connection.driver instanceof PostgresDriver &&
+    error.driverError.code === "40001"
+  ) {
     return true;
   }
 
   return false;
+};
+
+function errorWithStack(error: unknown, stack?: string) {
```

Comment:
why is the `stack` optional?

Reply:
Because the type of stack on the error object is type of string that is optional.